Tests/SwiftSyntaxBuilderTest/FunctionTests.swift

```diff
@@ -60,4 +60,43 @@ final class FunctionTests: XCTestCase {
       }
       """)
   }
+
+  func testParensArePreservedIfSet() {
+    let buildable = FunctionCallExpr(
+      "test",
+      leftParen: .leftParen.withLeadingTrivia(.spaces(1)),
+      rightParen: .rightParen.withTrailingTrivia(.spaces(1)),
+      trailingClosure: ClosureExpr())
+    let syntax = buildable.buildSyntax(format: Format())
+    XCTAssertEqual(syntax.description, "test () {}")
+  }
+
+  func testParensEmittedIfNoArgumentsAndTrailingClosures() {
+    let buildable = FunctionCallExpr("test")
+    let syntax = buildable.buildSyntax(format: Format())
+    XCTAssertEqual(syntax.description, "test()")
+  }
+
+  func testParensEmittedForArgumentAndTrailingClosure() {
+    let buildable = FunctionCallExpr("test", trailingClosure: ClosureExpr(), argumentListBuilder: {
+      TupleExprElement(expression: "42")
+    })
+    let syntax = buildable.buildSyntax(format: Format())
+    XCTAssertEqual(syntax.description, "test(42){}")
+  }
+
+  func testParensOmittedForTrailingClosure() {
+    let buildable = FunctionCallExpr("test", trailingClosure: ClosureExpr())
+    let syntax = buildable.buildSyntax(format: Format())
+    XCTAssertEqual(syntax.description, "test{}")
+  }
+
+  func testParensOmittedForMultipleTrailingClosures() {
+    let buildable = FunctionCallExpr("test", additionalTrailingClosuresBuilder: {
+      MultipleTrailingClosureElement(label: .identifier("a").withLeadingTrivia(.spaces(1)), closure: ClosureExpr())
+      MultipleTrailingClosureElement(label: .identifier("b").withLeadingTrivia(.spaces(1)), closure: ClosureExpr())
+    })
+    let syntax = buildable.buildSyntax(format: Format())
+    XCTAssertEqual(syntax.description, "test a: {} b: {}")
```

Comment:
This will not compile because the first trailing closure needs to be unlabeled. Could you change the test case to pass the first trailing closure as the `trailingClosure` argument instead of in the `additionalTrailingClosuresBuilder`?

Reply:
I think we shouldn't allow building invalid code, similarly to the case with no parenthesis.
Should we drop the `trailingClosure` argument and populate it from the `additionalTrailingClosuresBuilder`? Alternatively we might provide two initializers, one with just optional `trailingClosure`(no `additionalTrailingClosuresBuilder`) and another with both non-optional `trailingClosure` and `additionalTrailingClosuresBuilder`. What do you think?

Reply:
I don’t think we should try too hard to prevent users from constructing SwiftSyntax trees that won’t compile. They will always be able to construct such trees using the `SyntaxFactory` API. My standpoint was that everything that matches the constraints in `gyb_syntax_support` should be considered a valid SwiftSyntax tree.

I think the cleanest solution here would be to restructure `FunctionCallExpr` in `gyb_syntax_support` in a way that enforces that there is an unlabeled trailing closure in front of labelled trailing closures. That way we can also make the same assumption when consuming a SwiftSyntax tree. Such a change would be slightly bigger but still manageable IMHO.